### PR TITLE
Data migration: Add migration-overview.md

### PR DESCRIPTION
**Pull request by toutdesuite**
### What is changed, added or deleted? (Required)

Add migration-overview.md

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3363/, https://github.com/pingcap/docs-cn/pull/3292/
- Other reference link(s):

### CODE REVIEW

File: data-migration-route.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
```
Comment (yikeke) [inline data-migration-route.md:1]: Please update the TOC according to [docs-cn TOC](https://raw.githubusercontent.com/pingcap/docs-cn/master/TOC.md). @toutdesuite

**Comment by yikeke** [comment]
Please add the necessary labels. @toutdesuite

**Comment by yikeke** [comment]
@CaitinChen will help review this PR.

**Comment by CaitinChen** [comment]
Please fix CI issues @toutdesuite

**Comment by toutdesuite** [comment]
> Please fix CI issues @toutdesuite

CI fixed

File: data-migration-route.md (hunk 1 above)
Comment (toutdesuite) [reply data-migration-route.md]: update in 52f3c9142cbe360f7838a5685c90538de13771fb

**Comment by yikeke** [comment]
PTAL @WangXiangUSTC

File: migration-overview.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+#### Migration method
+
+1. Use Mydumper to export the full MySQL data.
+2. Use TiDB Lightning to import the full data to TiDB. For details, refer to [Migrate from Mydumper Files](/migrate-from-mysql-mydumper-files.md).
+
+### Use DM
+
+#### Scenarios
+
+You can use DM to migrate full MySQL data and to replicate incremental data. Note that the size of the full data must be less than 1 TB. Otherwise, it is recommended to use Mydumper and TiDB Lightning to import the full data, and then use DM to replicate the incremental data.
```
Comment (WangXiangUSTC) [inline migration-overview.md:33]: It is suggested that the full data less than 1TB, not must.

Comment (toutdesuite) [reply migration-overview.md]: updated in 73f64ae98590e8c63f441cdae81933d8cebab1bc

**Comment by toutdesuite** [comment]
@CaitinChen PTAL again, thanks

**Pull request merged**